app/src/main/java/cm/aptoide/pt/timeline/TimelineAnalytics.java

```diff
@@ -664,4 +693,203 @@ public void scrollToPosition(int position) {
             bodyInterceptor, httpClient, converterFactory, tokenInvalidator, appId,
             sharedPreferences));
   }
+
+  public HashMap<String, Object> parseEventData(CardTouchEvent event, boolean status, int errorCode){
```

Comment:
We need to think about a way to shorten this and the parseShareCompletedEventData methods.

Some duplicated code can be removed because there are common parts to all events like:
    data.put("position", event.getPosition());
    data.put("card_type", card.getType());

Maybe we should extract the error parsing switch out to an object that deals with this kind of logic.. on other hand most of this methods (also the others that already existed like sendClickOnPostBodyEvent) make me feel like there is logic that this entity should deal with. 

We should probably talk and try to think of any ideas when you reach this part.